src/app/register/register.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit } from '@angular/core';
+import { RegisterService } from '../services/register/register.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  username: string;
+  password: string;
+  confirmPsw: string;
+  usernameError: string;
+  passwordError: string;
+  confirmPswError: string;
+  messageSuccess: string;
+  messageFail: string;
+
+  constructor(private registerService: RegisterService, private router: Router) { }
+
+  ngOnInit() {
+  }
+
+  CheckUserName(username: string): boolean {
+    if (username !== undefined && username.length > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  checkUserExists(username: string): boolean {
```

Comment:
This should invoke a service call where you get the backend result on checking the user name if already exist.
You should mock the service call depending on what you expect in a response.

Reply:
solved